api/src/main/java/io/grpc/NameResolver.java

```diff
@@ -350,6 +350,11 @@ public final void onAddresses(
   @Documented
   public @interface ResolutionResultAttr {}
 
+  @Internal
+  @ResolutionResultAttr
+  public static final Attributes.Key<Map<String, ?>> RESOLUTION_RESULT_ATTR_HEALTH_CHECKING_CONFIG =
```

Comment:
This should be in `GrpcAttributes.java`

Reply:
in current world yes, but the `Attributes` is actually API not an implementation of `io.grpc.api` (core). for example, if user depends on `io.grpc.services` 1.n.x, `io.grpc.api` 1.n.x and `io.grpc.api` 1.(n-1).x. it won't work. because, we are using the attributes as API. so, going forward the shared attributes should be in the api not in core. we are trying to fix this issue slowly from this change.
@ejona86 can explain it better (or correct my understanding).

Reply:
`io.grpc.internal` should be considered convenience. Easiest test to check if it is convenience is to ask if it would work if each user of it shaded it. This attribute is used cross-module, between a transport/core and a load balancer, so it should not be in internal.

The same would be true of `GrpcAttributes.ATTR_LB_PROVIDED_BACKEND`. `ATTR_CLIENT_EAG_ATTRS` is a hack that maybe should be moved out as well, but maybe we want to replace it with something else. So that just leaves `ATTR_SECURITY_LEVEL`, which is sort of fine, but could be done in a more explicit way; it was sort of a lazy solution to pass data out of transports.

So in general, GrpcAttributes is a code smell and any new attributes added there should be treated with suspicion.